### PR TITLE
Fix typos, slightly improve code flow and stuff

### PR DESCRIPTION
**Pull request by SolalPirelli**
This was done in GitHub's online editor since I don't have my laptop with me right now, but it should compile.

### CODE REVIEW

File: watchpoints.c [hunk 1]
```diff
@@ -133,79 +126,71 @@ static long watchpoints_ioctl(struct file *file, unsigned int cmd, long
...
-			return -EINVAL;
-
+	case REMOVE_BREAKPOINT:
+		for(int i = 0; i < WATCHPOINTS_MAX; i++) {
+			if(watchpoints[i] &&
+			   watchpoints[i]->attr.bp_addr == data.data_ptr &&
+			   watchpoints[i]->ctx->task->pid == data.pid) {
+				unregister_hw_breakpoint(watchpoints[i]);
+			}
+		}
```
Comment (SolalPirelli) [inline watchpoints.c:173]: The lack of a break here seems like a bug; do you always want to return EINVAL when removing a breakpoint?

File: watchpoints.c [hunk 2]
```diff
@@ -215,8 +200,8 @@ watchpoints_read(struct file *file, char __user * user_buffer,
 		
 		while(length && *output_pointer) {
```
Comment (SolalPirelli) [inline watchpoints.c:205]: There has to be a smarter way to do it than put_user-ing every single character in the string... (I don't know about it though)

Comment (tharvik) [inline watchpoints.c:205]: Yes, there is, `copy_to_user` if I recall correctly.

File: watchpoints.c [hunk 3]
```diff
@@ -74,24 +75,20 @@ module_init(watchpoint_init);
...
+			       struct perf_sample_data *data, 
+			       struct pt_regs *regs)
 {
-	int i;
-	long size;
-	
-	for(i=0; i < WATCHPOINTS_MAX; i++) {
-		struct change_list *new_change;
-		
+	for(int i = 0; i < WATCHPOINTS_MAX; i++) {
```
Comment (BenjaminSchubert) [inline watchpoints.c]: please use C89 style, as it is most widely used in the kernel

**Comment by BenjaminSchubert** [comment]
Please don't change C89 coding style

File: watchpoints.c (hunk 3 above)
Comment (BenjaminSchubert) [inline watchpoints.c]: this is not c89

File: watchpoints.c [hunk 4]
```diff
@@ -133,79 +126,71 @@ static long watchpoints_ioctl(struct file *file, unsigned int cmd, long
...
-watchpoints_read(struct file *file, char __user * user_buffer,
-                 size_t length, loff_t * offset)
+static ssize_t watchpoints_read(struct file *file, char __user *user_buffer,
+                		size_t length, loff_t *offset)
 {
 	struct change_list *new_change;
-	char template[] = "pid=%d, pointer=%llu, value=%s\n";
-	char* output;
-	char* output_pointer;
+	char[] template = "pid=%d, pointer=%llu, value=%s\n";
```
Comment (BenjaminSchubert) [inline watchpoints.c]: please don't use java in C89 code

File: watchpoints.c (hunk 1 above)
Comment (BenjaminSchubert) [inline watchpoints.c:173]: Thanks a lot, I will correct this one.

File: watchpoints.c [hunk 5]
```diff
@@ -133,79 +126,71 @@ static long watchpoints_ioctl(struct file *file, unsigned int cmd, long
...
-			attr.bp_addr = data.data_ptr;
-			attr.bp_len = HW_BREAKPOINT_LEN_4;
-			attr.bp_type = HW_BREAKPOINT_W;
-
-			tsk = pid_task(find_vpid(data.pid), PIDTYPE_PID);
-
-			perf_watchpoint = register_user_hw_breakpoint(&attr,
-					watchpoint_handler, NULL, tsk);
+	case ADD_BREAKPOINT:
+		// Initialize breakpoint
```
Comment (SolalPirelli) [inline watchpoints.c]: This is C99 too, you know :p

File: watchpoints.c [hunk 6]
```diff
@@ -133,79 +128,74 @@ static long watchpoints_ioctl(struct file *file, unsigned int cmd, long
...
+		attr.bp_len = HW_BREAKPOINT_LEN_4;
+		attr.bp_type = HW_BREAKPOINT_W;
 
-			tsk = pid_task(find_vpid(data.pid), PIDTYPE_PID);
+		tsk = pid_task(find_vpid(data.pid), PIDTYPE_PID);
 
-			perf_watchpoint = register_user_hw_breakpoint(&attr,
-					watchpoint_handler, NULL, tsk);
+		struct perf_event *perf_watchpoint = 
+			register_user_hw_breakpoint(&attr, watchpoint_handler, NULL, tsk);
```
Comment (BenjaminSchubert) [inline watchpoints.c]: C89 style please

File: watchpoints.c [hunk 7]
```diff
@@ -133,79 +128,74 @@ static long watchpoints_ioctl(struct file *file, unsigned int cmd, long
...
+				watchpoints[i] = perf_watchpoint;
+				watchpoint_data_size[i] = data.data_size;
+				break;
 			}
+		}
+		break;
+	}
+	
+	case REMOVE_BREAKPOINT:
+		for(i = 0; i < WATCHPOINTS_MAX; i++) {
```
Comment (BenjaminSchubert) [inline watchpoints.c:167]: i is not in the same scope now, please correct

File: watchpoints.c [hunk 8]
```diff
@@ -213,44 +202,43 @@ watchpoints_read(struct file *file, char __user * user_buffer,
 		output_pointer = output + last_entry_offset;
 		printk(KERN_DEBUG "%s", output);
 		
+		bytes_read = 0;
```
Comment (BenjaminSchubert) [inline watchpoints.c]: You introduced a bug here. bytes_read should be set to 0 before the outer loop. please correct this

File: watchpoints.c [hunk 9]
```diff
@@ -213,44 +202,43 @@ watchpoints_read(struct file *file, char __user * user_buffer,
...
 	return bytes_read;
-	
 }
 
 
 static int __init watchpoint_init(void)
 {
 	void *ptr_err;
-
+
```
Comment (BenjaminSchubert) [inline watchpoints.c]: Please do not add spaces at the end of lines

Comment (SolalPirelli) [inline watchpoints.c]: Tell that to GitHub's online editor :p

File: README.md [hunk 10]
```diff
@@ -1,15 +1,15 @@
...
 
 ## Usage
 
 #### Compilation
-To compile watchpoint, you only have to hit `make` in the top directory
+To compile watchpoints, hit `make` in the top directory.
 
 #### Loading
-watchpoints is loaded as a normal kernel module as : `insmod ${path_to_watchpoints.ko}`
+watchpoints is loaded as a normal kernel module as `insmod ${path_to_watchpoints.ko}` .
```
Comment (BenjaminSchubert) [inline README.md]: Adding a "." needs an uppercase letter at the beginning

File: README.md [hunk 11]
```diff
@@ -1,15 +1,15 @@
 # watchpoints
-A Linux kernel module for setting up watchpoints without going through ptrace
+A Linux kernel module for setting up watchpoints without going through ptrace.
```
Comment (BenjaminSchubert) [inline README.md]: No "." at the end of a statement

File: watchpoints.c [hunk 12]
```diff
@@ -49,14 +49,15 @@ struct perf_event *watchpoints[WATCHPOINTS_MAX];
...
 
-static void watchpoint_handler(struct perf_event *bp, struct perf_sample_data
-			       *data, struct pt_regs *regs);
-static long watchpoints_ioctl(struct file *file, unsigned int cmd, long
-			      unsigned ptr_message);
+static void watchpoint_handler(struct perf_event *bp, 
+			       struct perf_sample_data *data, 
+			       struct pt_regs *regs);
+static long watchpoints_ioctl(struct file *file, unsigned int cmd, 
+			      long unsigned ptr_message);
```
Comment (BenjaminSchubert) [inline watchpoints.c]: When aligning arguments to the function, please align under the first argument and not the opening parenthesis

File: watchpoints.c [hunk 13]
```diff
@@ -250,7 +237,7 @@ static int __init watchpoint_init(void)
 	}
 
 	ptr_err = device_create(watchpoints_class, NULL, MKDEV(MAJOR_NUM, 0),
-				NULL, DEVICE_NAME);
+				      NULL, DEVICE_NAME);
```
Comment (BenjaminSchubert) [inline watchpoints.c]: why moving this like that ?

File: README.md [hunk 14]
```diff
@@ -1,15 +1,15 @@
...
 
 ## Usage
 
 #### Compilation
-To compile watchpoint, you only have to hit `make` in the top directory
+To compile watchpoints, hit `make` in the top directory.
 
 #### Loading
-watchpoints is loaded as a normal kernel module as : `insmod ${path_to_watchpoints.ko}`
+Watchpoints is loaded as a normal kernel module as `insmod ${path_to_watchpoints.ko}` .
```
Comment (BenjaminSchubert) [inline README.md]: No space before a "." please

**Comment by BenjaminSchubert** [comment]
Too much changes, nothing compiles. No thanks.

I will still take the readme and change the line where it's worth it. Thanks for that.

**Pull request closed without merging**